app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -245,7 +243,7 @@ def server_status
     def facts
       User.as_anonymous_admin do
         sync_task(::Actions::Katello::Host::Update, @host, rhsm_params)
-        @host.subscription_facet.update_facts(rhsm_params[:facts]) unless rhsm_params[:facts].blank?
+        Katello::Host::SubscriptionFacet.update_facts(@host, rhsm_params[:facts])
```

Comment:
In the code you linked, where is the host created? At the save (https://github.com/Katello/katello/pull/5713/files#diff-50713e26c68a3af05393d936d0f19328R25) ?